docs(svelte-testing-library): update vitest and jest setup instructions (#1360)
testing-library/testing-library-docs: docs/svelte-testing-library/setup.mdx

sidebar_label: Setup
---

We recommend using [Vitest][], but you're free to use the library with any test
runner that's ESM compatible.

[vitest]: https://vitest.dev/

## Vitest

1. Add development dependencies

   - [`@testing-library/svelte`][@testing-library/svelte]
   - [`@testing-library/jest-dom`][@testing-library/jest-dom] (Optional, but
     highly recommended)
   - [`@sveltejs/vite-plugin-svelte`][@sveltejs/vite-plugin-svelte]
   - [`vitest`][vitest]
   - [`jsdom`][jsdom], [`happy-dom`][happy-dom], or other [Vitest DOM
     environment][]

   ```bash npm2yarn
   npm install --save-dev \
     @testing-library/svelte \
     @testing-library/jest-dom \
     @sveltejs/vite-plugin-svelte \
     vitest \
     jsdom
   ```

   Optionally install [`@vitest/ui`][@vitest/ui], which opens a UI within a
   browser window to follow the progress and interact with your tests.

   ```bash npm2yarn
   npm install --save-dev @vitest/ui
   ```

> 2. Add a `vitest-setup.js` file to configure cleanup and
>    [`@testing-library/jest-dom`][@testing-library/jest-dom], if using.

Reviewer: ```suggestion
2. Add a `vitest-setup.js` file to optionally set up automatic post-test cleanup and 
   [`@testing-library/jest-dom`][@testing-library/jest-dom] expect matchers.
```

>    ```ts
>    // vitest-setup.js

Reviewer 2: Could you use the title's feature on this page instead of an inline file names?
https://docusaurus.io/docs/2.x/markdown-features/code-blocks#code-title

For example:
```suggestion
   ```ts title="vitest-setup.js"
```

   import '@testing-library/svelte/vitest'
   import '@testing-library/jest-dom/vitest'
   ```

3. Add `vitest.config.js`, or update your existing `vite.config.js`, to process
   Svelte files, resolve browser exports during tests, use the [jsdom][] (or
   [happy-dom][]) environment, and run your setup file.

   ```js
   // vitest.config.js
   import {defineConfig} from 'vitest/config'
   import {svelte} from '@sveltejs/vite-plugin-svelte'

   export default defineConfig(({mode}) => ({
     plugins: [svelte()],
     resolve: {
       conditions: mode === 'test' ? ['browser'] : [],
     },
     test: {
       environment: 'jsdom',
       setupFiles: ['./vitest-setup.js'],
     },
   }))
   ```

   :::note

   Prepending the `browser` resolve condition to Vite's default conditions may
   cause issues if you have a complex Vite configuration or dependencies that
   cannot be loaded into Node.js

   See [testing-library/svelte-testing-library#222][] for more information and
   alternative configuration options to ensure Svelte's browser bundle is used.
   :::

> 4. Add test scipts to your `package.json` to run the tests with Vitest

Reviewer: ```suggestion
4. Add test scripts to your `package.json` to run the tests with Vitest
```
